### DISCUSSION TITLE
Cursor position missing in vim mode with markdown syntax highlighting

### DISCUSSION THREAD

**Discussion by kyletolle** [standardnotes/forum#2459] [General] [answered]
I appreciate Standard Notes and all the hard work that goes into improving it!

Following up with an issue I initially reported on twitter as advised to here: https://twitter.com/StandardNotes/status/1078742198387126274

## Overview

When using vim mode with markdown highlighting, moving through the document in normal mode, I lose the cursor highlighting when navigating through Markdown code blocks, which makes it nearly impossible to jump around and edit the text.

## Steps to reproduce

- Use macOS Mojave 10.14.1
- Use SN 3.0.1
- Create note using the Vim editor
- Change language to `Markdown` for the syntax highlighting
- Enter vim's insert mode (hit `i`)
- Type a Markdown code block (using either 3 backticks or indent)
- Enter vim's normal mode (hit `Esc`)

## Actual behavior

Cursor position within the markdown code block is hidden. I cannot see the cursor at all. This means I have no idea where the cursor is when I'm trying to navigate around and type.

I'm guessing the code block's background color is obscuring the cursor highlight.

## Expected behavior

Should see the cursor highlighted. So you know where the cursor is and where you're navigating. through the text.

## Screenshots

- Using vim editor
![vim editor](https://user-images.githubusercontent.com/260869/50534056-a22acf00-0b04-11e9-9272-59bfcb3b7ba6.png)
- Using markdown language
![markdown language](https://user-images.githubusercontent.com/260869/50534060-b40c7200-0b04-11e9-9b98-e9233a180b08.png)
- What cursor highlight looks like when not in a code block
  - Notice the green box that indicates where the cursor is
![green cursor](https://user-images.githubusercontent.com/260869/50533992-92f75180-0b03-11e9-93d2-23dfb33d6aaa.png)
- No cursor highlight when cursor is within the codeblock
  - My cursor is actually at the beginning of "What" but there's no way to see that
![no cursor](https://user-images.githubusercontent.com/260869/50534027-131db700-0b04-11e9-8689-a14306ed4738.png)
- What cursor looks like in code block after `cw`
  - I've done a `cw` to start replacing the "What" word, and you can see a white background near where the cursor is
![white cursor](https://user-images.githubusercontent.com/260869/50533997-aaced580-0b03-11e9-80de-17a4b2f8733f.png)
- Cursor is visible when there's whitespace at the beginning of the line
  - Here I typed a couple spaces before the word, hit `Esc` to get back to normal mode, and can see the cursor now, because the code block background highlighting doesn't apply here
![green cursor again](https://user-images.githubusercontent.com/260869/50534099-896ee900-0b05-11e9-8cd7-e2a1639ca999.png)
- The cursor is visible when in insert mode and typing
  - I think this is normal and expected. It's a blinking line instead of a blinking rectangle.
![blue cursor](https://user-images.githubusercontent.com/260869/50534082-064d9300-0b05-11e9-8682-8582ba09fa93.png)

---

I hope the steps, details, and screenshots help. Please let me know if I can provide any additional information!

**Comment by atmoio** [comment]
Hey @kyletolle, thanks for the detailed report! It looks like the cursor is hiding behind the gray code blocks in Markdown mode. This should now be fixed in version 1.2.3 of the Vim editor. You can update via the Extensions menu. On web, be sure to clear your cache if the update doesn't seem to have taken effect.

**Comment by kyletolle** [comment]
Thanks for the quick fix @mobitar! With Vim Editor v1.2.3 I can see the cursor again. One more question. Should the character be visible when the cursor is highlighted?

In vim itself, I see something like this:
![vim cursor and character highlight](https://user-images.githubusercontent.com/260869/50567452-f7d4c680-0d12-11e9-9edd-ace1917dd83c.png)
When the cursor is visible, the character gets an inverse color.

But in SN, I see this:
![SN cursor highlight](https://user-images.githubusercontent.com/260869/50567466-1c30a300-0d13-11e9-8b59-7eb29d41acfa.png)

When the cursor is visible in SN's vim editor, I can't see the character the cursor is on.

**Comment by kyletolle** [comment]
And I'm happy to have given the detailed report. I know how providing the information upfront allows you to get right into fixing the code instead of having to follow up with me about missing information.

**Comment by kyletolle** [comment]
Similarly, when I am in insert mode and double click to select text within a markdown codeblock, the background color for the highlighted text is obscured by the codeblock background color.

**Comment by atmoio** [comment]
Hmm, good point. Have to catch up on some other stuff so won't be able to get a quick fix, but have made a note to tackle this next time I'm in this vicinity.

**Comment by kyletolle** [comment]
Text highlighting also seems to be obscured when search terms are highlighted in a note.

**Comment by myreli** [comment, marked as answer]
As of version `3.149.5` all reported issues seem to be resolved. 

We are currently triaging our older issues, please let us know if you are still facing any problems.
